core/domain/opportunity_services.py

```diff
@@ -386,3 +405,65 @@ def update_opportunities_with_new_topic_name(topic_id, topic_name):
 
     _save_multi_exploration_opportunity_summary(
         exploration_opportunity_summary_list)
+
+
+def regenerate_opportunities_related_to_topic(
+        topic_id, delete_existing_opportunities=False):
+    """Regenerates opportunity models which belongs to a given topic.
+
+    Args:
+        topic_id: str. The ID of the topic.
+        delete_existing_opportunities: bool. Whether to delete all the existing
+            opportunities related to the given topic.
+
+    Returns:
+        dict. A dict representing the result of the regeneration process.
```

Comment:
No, this shouldn't be a dict. Make a domain object for it, or return a tuple with fixed types.

One alternative thing you could perhaps do is to raise an error and let the caller handle it appropriately.

Reply:
Earlier this functionality was only in the jon_one_off file so I had a dict then, now I had to create a method so that I can reuse it in one_off job and in the admin tab, so I just moved the functionality into this method without realizing it's in the service layer. I've added an exception instead. Done!